### DISCUSSION TITLE
Screen for public users to create rows (app access is Basic logged in user or above)

### DISCUSSION THREAD

**Discussion by saz33m1** [Budibase/budibase#14971] [Help] [answered]
Using the tutorial sample data, I have an employees table (with access set to Basic logged in users). I want a public form (screen) to be able to submit rows. 

I tried the solution detailed in https://github.com/Budibase/budibase/discussions/10563, however, I don't want public users to read my **employees** table, and only be able to create rows. 

Right now, I am getting "Session Unauthenticated" error when viewing the the public form (screen) as an unauthenticated user. 

Any help is appreciated it.

<img width="810" alt="image" src="https://github.com/user-attachments/assets/08323630-d738-4dad-bb20-085dca358296">

<img width="785" alt="image" src="https://github.com/user-attachments/assets/09c2bc5b-fe3f-4e16-8013-3c176a9b4819">

<img width="645" alt="image" src="https://github.com/user-attachments/assets/99ecf7a3-1abd-44e0-89fe-b42580296bac">

**Comment by saz33m1** [comment]
Using the the latest release 3.2.1. When I try to bring up the public form, it displays, but shows an error and won't let me submit (create new row in table set to private access).
![Screenshot 2024-11-08 150144](https://github.com/user-attachments/assets/d541ff95-d221-41ab-9e97-ba13b05bcbb4)

**Comment by mikesealey** [comment]
Hi @saz33m1 

The way permissions operate has been worked on recently and updated in last week's Budibase v3 Launch Week. This means that if the information discussed in May 2023 does not apply, it will likely be outdated.

When setting a table's permissions to "Public" the table can be written to from a public screen in your app. If you are still encountering the "Session not authenticated" error message this may be a bug and you should submit an issue here: https://github.com/Budibase/budibase/issues/new/choose

I have just tested this, and a Public table can be written to from a public form. There is an ongoing discussion within the engineering teams regarding public-write-logged-in-read permissions. There are currently no feature requests for this issue, you may wish to submit a feature request using the link above. You could provide more specifics on your intended use-case which may help to shape any future updates in this area.

I will be meeting with the rest of the team throughout the week and we could discuss your feature request further.
Mike

**Comment by SimonDoll** [comment]
HI @mikesealey thanks a lot for looking into that!

I'm currently facing the exact same issue / missing feature of public-write-logged-in-read which was possible with Budibase < 3.0
If it helps, I could also submit a feature request @saz33m1 / upvote Yours when it exists.

A prototypical use-case is nicely illustrated in one of your templates: visitor-check-in-form template @mikesealey (see #14973).
In my case, we have a lot of "single-time" customers and a restricted timeframe.
Therefore, the customers just enter some details about themselves and submit their cases, but should never have access to other cases and don't need to change their case. Also, we don't want them to go through a register process since they will most likely not return / time is limited.

Sadly, the new permissions currently prevent me from migrating my apps to Budibase>=3.0

I hope that helps and that @mikesealey and the team find a solution 👍

**Reply by eumelz** [reply]
I made a feature request: https://github.com/Budibase/budibase/discussions/15892 :*

**Comment by andz-bb** [comment, marked as answer]
I've written a tutorial for this in the Budibase Docs. You can find it here: https://docs.budibase.com/docs/public-form

**Reply by eumelz** [reply]
Thanks for the tutorial!

Though I need to say, this is very tedious to implement when you have many fields in your Form. And it's so many places you need to change things if you add a field or so. That's not the spirit of Budibase at all! It used to be sooo easy to make a public form. Why don't you change the table level access control to have 3 Options: Read, Add, Update and have each one assign a role. 

I'm thinking about skipping the Automation workaround to wait for an update to make this way easier. Do you have any idea if and when you'll update this?

also @mikesealey 

Thanks a lot ❤

**Reply by eumelz** [reply]
@mikesealey Do you have any Updates on this? :)

**Reply by M4s0n333** [reply]
Hi @mikesealey , I'd also be very interested in an update. The above functionality @eumelz would be perfect. This would significantly simplify working with Budibase, as the workflow using automations is too time-consuming.

**Reply by eumelz** [reply]
I made a feature request: https://github.com/Budibase/budibase/discussions/15892

**Reply by mikesealey** [reply]
@eumelz @M4s0n333 @SimonDoll 

I was discussing this with m'colleague, @ConorWebb96 who proposed a much simpler way to achieve this -

On your chosen table, with the permissions set to App-User or above, create a view
![image](https://github.com/user-attachments/assets/61943d91-9bf0-4e22-9df4-9382eb397c21)

In the view, set the permissions to Public, and write a filter that will never be true. In my case I'm filtering for rows with an ID of banana. My filters will return no rows if there are no matches.

![image](https://github.com/user-attachments/assets/fb02a7b4-36b1-4b69-a170-eb9e3b1aae55)

The result of this is that when the public-write view is used, no rows are returned, but it can be used to write.

**Reply by eumelz** [reply]
Hey, thanks a lot for this solution! I hope it's bullet proof regarding security, since I've just implemented it in prod. 
I think many users would appreciate if you updated this tutorial: https://docs.budibase.com/docs/public-form
